torchsynth/module.py

```diff
@@ -622,62 +622,39 @@ class Noise(SynthModule):
 
     # Do we really want deterministic noise within each batch?
     # https://github.com/torchsynth/torchsynth/issues/250
-    noise_batch_size: int = BATCH_SIZE_FOR_REPRODUCIBILITY
+    noise_batch_size: int = BASE_REPRODUCIBLE_BATCH_SIZE
     # Unfortunately, Final is not supported until Python 3.8
     # noise_batch_size: Final[int] = BATCH_SIZE_FOR_REPRODUCIBILITY
 
     def __init__(self, synthconfig: SynthConfig, seed: int, **kwargs):
         super().__init__(synthconfig, **kwargs)
 
-        # https://github.com/torchsynth/torchsynth/issues/255
-        if (
-            self.batch_size != BATCH_SIZE_FOR_REPRODUCIBILITY
-            and self.synthconfig.reproducible
-        ):
-            raise ValueError(
-                "Reproducibility currently only supported "
-                f"with batch_size = {BATCH_SIZE_FOR_REPRODUCIBILITY}. "
-                "If you want a different batch_size, "
-                "initialize SynthConfig with reproducible=False"
-            )
-
         # Pre-compute default batch size number of noise samples
         generator = torch.Generator(device="cpu").manual_seed(seed)
-        noise = torch.empty((self.noise_batch_size, self.buffer_size), device="cpu")
-        noise.data.uniform_(-1.0, 1.0, generator=generator)
 
-        # Repeat the noise batches if the current batch size is larger than the default
-        if self.batch_size > self.noise_batch_size:
+        # In reproducible mode, we support batch sizes that are multiples
+        # of the BASE_REPRODUCIBLE_BATCH_SIZE
+        if self.synthconfig.reproducible:
             if self.batch_size % self.noise_batch_size != 0:
                 raise ValueError(
-                    "Batch size is not divisible by the default "
-                    f"batch size of {self.noise_batch_size}"
+                    f"Batch size must be a multiple of {self.noise_batch_size} "
+                    "when using reproducible mode. Either change your batch size,"
+                    "or set reproducible=False in the SynthConfig for this module."
                 )
 
-            noise = noise.repeat(self.batch_size // self.noise_batch_size, 1)
+            noise = torch.empty((self.noise_batch_size, self.buffer_size), device="cpu")
+            noise.data.uniform_(-1.0, 1.0, generator=generator)
+            if self.batch_size > self.noise_batch_size:
+                noise = noise.repeat(self.batch_size // self.noise_batch_size, 1)
```

Comment:
I'm a bit scared of this lol, did you write a test? Maybe it's later in the PR review. I haven't used .repeat much

Reply:
Don't see a test can we write one specifically for noise? I know we have one for synth but it would be nice

Reply:
Yup we have one -- made a comment above to point it out

Reply:
Already had one that tested noise reproducibility -- updated it slightly in this PR to make sure reproducible mode is on